src/lvm.c

```diff
@@ -1135,6 +1138,22 @@ void luaV_finishOp (lua_State *L) {
 #define vmcase(l)	case l:
 #define vmbreak		break
 
+#ifdef WITH_YK
+/*
+ * Given a pointer to a Lua instruction in the instruction stream, return (if
+ * possible) a non-NULL pointer to a the corresponding `YkLocation`, or `NULL`
+ * if there is no corresponding `YkLocation`.
+ */
+static YkLocation *luaV_lookup_ykloc(const Instruction *pc, const CallInfo *ci) {
```

Comment:
I think this should be in the main loop: in this form we're implicitly hoping the optimiser will inline this function, notice that there are shared pointers, etc etc.

Reply:
Can we mark the function `__attribute__((always_inline))` so as to not clutter up the main loop?

Reply:
That would solve inlining, but you're still relying on the optimiser to optimise away all the loads etc. This really would be better manually inlined IMHO.

Reply:
I think it junks up the main loop. If we were to follow precedent, it'd be an unhygienic macro, like `vm_fetch()`:

```
/* fetch an instruction and prepare its execution */                                                                                                                                                               
#define vmfetch()   { \                                                                                                                                                                                            
  if (l_unlikely(trap)) {  /* stack reallocation or hooks? */ \                                                                                                                                                    
    trap = luaG_traceexec(L, pc);  /* handle hooks */ \                                                                                                                                                            
    updatebase(ci);  /* correct stack */ \                                                                                                                                                                         
  } \                                                                                                                                                                                                              
  i = *(pc++); \                                                                                                                                                                                                   
  ra = RA(i); /* WARNING: any stack reallocation invalidates 'ra' */ \                                                                                                                                             
}                                                     
```

Reply:
An unhygienic macro is fine with me. Performance here is critical.

Reply:
Fixed in 4e8c772b5149